src/libstore/build/local-derivation-goal.cc

```diff
@@ -720,6 +722,15 @@ void LocalDerivationGoal::startBuilder()
             chownToBuilder(*cgroup);
             chownToBuilder(*cgroup + "/cgroup.procs");
             chownToBuilder(*cgroup + "/cgroup.threads");
+            auto parentCgroup = dirOf(*cgroup);
+            writeFile(parentCgroup + "/cgroup.subtree_control", "+memory");
```

Comment:
This will fail if the cgroup does not have `memory` in `cgroup.controllers`, or if it's not a leaf cgroup or one that only has `pids` in its children (see the "no internal processes" rule in `cgroups(7)`). In particular, this may cause non-root chroot builds to fail (i.e. `--store /path/to/my/nix` as a non-root user) since we can't count on `memory` being enabled.

It's not clear what the best solution is, but some options:

* When `nix-daemon` starts, it can create a sub-cgroup (e.g. `system.slice/nix-daemon.service/manager`) for itself and move into it, and create sub-cgroups for the builds in the original cgroup (e.g. `system.slice/nix-daemon.service/<build>`). That way the original cgroup doesn't violate the "no internal processes" rule.
* That doesn't work for the non-root case, since the original cgroup will likely have other processes in it (e.g. `user.slice/user-1000.slice/user@1000.service/app.slice/app-org.kde.konsole-7edbbf6bbf864ef9a259f86f31b0fd90.scope`). So we can't easily convert it into a cgroup that doesn't violate the "no internal processes" rule. I guess we could search upwards for the nearest parent that has the `memory` controller enabled and has no internal processes (e.g. `/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/app.slice` in this example) but that's a bit naughty...
* Allow the parent cgroup for builds to be specified in `nix.conf`.

Whatever the solution, Nix shouldn't fail if writing to `cgroup.subtree_control` fails.